provider/docker/docker.go

```diff
@@ -153,15 +153,18 @@ func (p *Provider) Provide(configurationChan chan<- types.ConfigMessage, pool *s
 			if p.Watch {
 				ctx, cancel := context.WithCancel(ctx)
 				if p.SwarmMode {
+					errChan := make(chan error)
 					// TODO: This need to be change. Linked to Swarm events docker/docker#23827
 					ticker := time.NewTicker(SwarmDefaultWatchTime)
 					pool.Go(func(stop chan bool) {
+						defer close(errChan)
```

Comment:
I agree with @tanyadegurechaff on this. But it would be better to use this syntax instead:
```go
if err, ok := <-errChan; ok {
  return err
} else {
  // channel closed
  ...
}
```